cli/src/commands/rebase.rs

```diff
@@ -432,21 +441,42 @@ fn rebase_revision(
         })
         .try_collect()?;
 
-    // Finally, it's safe to rebase `old_commit`. At this point, it should no longer
-    // have any children; they have all been rebased and the originals have been
-    // abandoned.
-    rebase_commit(settings, tx.mut_repo(), &old_commit, &new_parents)?;
-    debug_assert_eq!(tx.mut_repo().rebase_descendants(settings)?, 0);
-
-    if num_rebased_descendants > 0 {
+    // Finally, it's safe to rebase `old_commit`. We can skip rebasing if it is
+    // already a child of `new_parents`. Otherwise, at this point, it should no
+    // longer have any children; they have all been rebased and the originals
+    // have been abandoned.
+    let skipped_commit_rebase = if old_commit.parents() == new_parents {
         writeln!(
             ui.stderr(),
-            "Also rebased {num_rebased_descendants} descendant commits onto parent of rebased \
-             commit"
+            "Skipping rebase of commit {}",
+            short_commit_hash(old_commit.id())
         )?;
+        true
```

Comment:
I used `workspace_command.commit_summary_template()` above, but I couldn't use it here since it is borrowed mutably. Not sure how to handle that.

Reply:
try `tx.write_commit_summary()` (or `tx.base_workspace_helper()` if WorkspaceCommandHelper is needed and the commit to be rendered by the template isn't mutated.)

Reply:
Thanks for the tip @yuja! I need to get more familiar with the codebase 😄